### DISCUSSION TITLE
Help with detct_planar_patches method

### DISCUSSION THREAD

**Discussion by coderPE** [isl-org/Open3D#6233] [Q&A]
**Background** - complete beginner with open3d or similar libraries. Trying to group/ classify 3d data (nodes, edges, surfaces). Experienced with 3d CAD/FEM geometries.

**Overall goal** - to group/ classify objects in 3D. 
**Intermediate goal** - Detect elements (points/lines/surfaces) that are in (or close to) a plane in a 3D CAD file.

**Current setup** - 
I am trying to implement detect_planar_patch method from open3d. I am getting the expected results for the example given on the website. 
![image](https://github.com/isl-org/Open3D/assets/32369174/56299f0c-3c86-47ab-af93-a49ba23f91ca)

My point cloud is generated from a mesh (in the .stl file) that contains 6 planar surfaces (so I am expecting 6 planes to be detected)

Source file (The .stl file is attached below)
![image](https://github.com/isl-org/Open3D/assets/32369174/df1431f0-9b64-43ba-81bf-e60008f1ee9f)
Generated point cloud using
```
mesh = o3d.io.read_triangle_mesh(path to .stl file)
pointcloud = mesh.sample_points_uniformly(1000)
```
![image](https://github.com/isl-org/Open3D/assets/32369174/1d8b0f44-9acf-40b8-b588-738f2ee27628)

This is the code in the tutorial

```
ply_point_cloud = o3d.data.PLYPointCloud()
pcd = o3d.io.read_point_cloud(ply_point_cloud.path)
# dataset = o3d.data.PCDPointCloud()
# pcd = o3d.io.read_point_cloud(dataset.path)
# pcd = pointcloud
# pcd = pcl

assert (pcd.has_normals())

# using all defaults
oboxes = pcd.detect_planar_patches(
    normal_variance_threshold_deg=60,
    coplanarity_deg=75,
    outlier_ratio=0.75,
    min_plane_edge_length=0,
    min_num_points=0,
    search_param=o3d.geometry.KDTreeSearchParamKNN(knn=30))

print("Detected {} patches". Format(len(oboxes)))

geometries = []
for obox in oboxes:
    mesh = o3d.geometry.TriangleMesh.create_from_oriented_bounding_box(obox, scale=[1, 1, 0.0001])
    mesh.paint_uniform_color(obox.color)
    geometries.append(mesh)
    geometries.append(obox)
geometries.append(pcd)

o3d.visualization.draw_geometries(geometries,
                                  zoom=0.6,
                                  front=[0.4361, -0.2632, -0.8605],
                                  lookat=[2.4947, 1.7728, 1.5541],
                                  up=[-0.1726, -0.9630, 0.2071])
```

But for my input cloud/geometry it doesn't detect any planes/ patches. 
**Things that I have tried:** 
1. Increasing point density from 1000 to 100000
2. Adjusting arguments for detect_planar_patches method

**Need help with:** 

1. I'm looking for help with a deeper dive with the arguments of the detect_planar_patches method. 
2. Open to suggestions with or without use of open3d library, but preferably in python.

[Sample model 1.zip](https://github.com/isl-org/Open3D/files/11875236/Sample.model.1.zip)

**Comment by rroell** [comment]
I think the zoom is not correct, since the position of the pointcloud is different to the example pointcloud.
Try without the window parameters. So:
```
o3d.visualization.draw_geometries(geometries,
                                  #zoom=0.6,
                                  #front=[0.4361, -0.2632, -0.8605],
                                  #lookat=[2.4947, 1.7728, 1.5541],
                                  #up=[-0.1726, -0.9630, 0.2071])
```

**Reply by hhmaizi** [reply]
voxel size matter and planar size matters, I think you should carefully choose a voxel_size to down sample your point cloud, and the planar should not be too small relative to your voxel_size. I mean your planars should contains enough points according to your voxel_size.
